lib/test.js

```diff
@@ -157,15 +159,55 @@ function domainError (er) {
   this.owner.threw(er)
 }
 
+// Called when endAll() is fired and there's stuff in the queue
+Test.prototype._queueFail = function () {
+  var queue = this._queue
+  this._queue = []
+  var len = queue.length
+  queue.forEach(function (q) {
+    var what = q[0]
+    var msg = what + ' left in queue'
+    var extra = { at: null }
+    if (what === 'test') {
+      extra = q[2]
+      if (q[1])
+        extra.name = q[1]
+      this.fail('child test left in queue', extra)
+    } else if (what === 'printResult') {
+      if (q[2] === 'test left unfinished: no end(), no plan()')
+        return
+      var msg = (q[1] ? 'ok' : 'not ok') + ' - ' + q[2].trim()
+      q[3].at = q[3].at || null
+      this.fail('test point left in queue: ' + (msg || '(no message)') , q[3])
+    } else if (what === 'spawn') {
+      extra = q[5]
+      extra.command = q[1]
+      extra.args = q[2]
+      extra.options = q[3],
+      extra.name = q[4]
+      this.fail('spawn left in queue', extra)
+    } else if (what === 'end') {
+      return
+    } else {
+      this.fail('left in queue: ' + what, {
+        at: this._calledAt
+      })
+    }
+  }, this)
+}
+
 Test.prototype.endAll = function () {
   var child = this._currentChild
+
+  if (this._queue && this._queue.length)
+    this._queueFail()
+
   if (child) {
     child._queue = []
-    if (!child._ended && !child._plan && child.fail) {
+    if (!child._ended && -1 === child._plan && child.fail)
```

Comment:
Note the bug here: `!child._plan` isn't the right test.  Before the child has a plan, it's `t._plan` is set to -1!